controllers/csi/driver/server.go

```diff
@@ -225,21 +226,29 @@ func (svr *CSIDriverServer) NodePublishVolume(ctx context.Context, req *csi.Node
 		targetPath,
 		Mount{Source: agentDir, Target: targetPath, ReadOnly: true},
 		Mount{
-			Source: filepath.Join(agentDir, "agent", "conf"),
-			Target: filepath.Join(targetPath, "agent", "conf"),
+			Source: filepath.Join(agentDir, dtcsi.AgentConfDir),
+			Target: filepath.Join(targetPath, dtcsi.AgentConfDir),
 		},
 		Mount{
-			Source: filepath.Join(envDir, "log", podUID),
-			Target: filepath.Join(targetPath, "log"),
+			Source: filepath.Join(envDir, dtcsi.LogDir, podUID),
+			Target: filepath.Join(targetPath, dtcsi.LogDir),
 		},
 		Mount{
-			Source: filepath.Join(envDir, "datastorage", podUID),
-			Target: filepath.Join(targetPath, "datastorage"),
+			Source: filepath.Join(envDir, dtcsi.DatastorageDir, podUID),
+			Target: filepath.Join(targetPath, dtcsi.DatastorageDir),
 		},
 	); err != nil {
 		return nil, status.Error(codes.Internal, fmt.Sprintf("failed to mount OneAgent volume: %s", err.Error()))
 	}
 
+	gcFile := filepath.Join(envDir, "gc", string(ver), podUID)
+	if err = ioutil.WriteFile(gcFile, nil, 0770); err != nil {
+		return nil, status.Error(codes.Internal, fmt.Sprintf("Failed to create file for garbage collector - error: %s", err))
+	}
+	if err = ioutil.WriteFile(filepath.Join(dtcsi.GarbageCollectionPath, volID), []byte(gcFile), 0770); err != nil {
+		return nil, status.Error(codes.Internal, fmt.Sprintf("Failed to create link file for garbage collector - error: %s", err))
+	}
```

Comment:
this creates an empty file at `/tmp/data/<env>/gc/<ver>/<podUID>` and copys the content of the file to `/gc/<volID>`. First of all, I think the target path should be `/tmp/gc/<volID>` if I understood that correctly. Secondly, the error message states that it couldn't link the file, but symlinking is done with `os.Symlink(...)`, wouldn't the above create a copy of the source file?